sdk/storage/azure-storage-blob/azure/storage/blob/_download.py

```diff
@@ -491,33 +512,17 @@ def _initial_request(self):
             except HttpResponseError:
                 pass
 
-        # If the file is small, the download is complete at this point.
-        # If file size is large, download the rest of the file in chunks.
-        # For encryption V2, calculate based on size of decrypted content, not download size.
-        if is_encryption_v2(self._encryption_data):
-            self._download_complete = len(self._current_content) >= self.size
-        else:
-            self._download_complete = response.properties.size >= self.size
-
         if not self._download_complete and self._request_options.get("modified_access_conditions"):
             self._request_options["modified_access_conditions"].if_match = response.properties.etag
 
         return response
 
-    def _get_downloader_start_with_offset(self):
-        # Start where the initial request download ended
-        start = self._initial_range[1] + 1
-        # For encryption V2 only, adjust start to the end of the fetched data rather than download size
-        if self._encryption_options.get("key") is not None or self._encryption_options.get("resolver") is not None:
-            start = (self._start_range or 0) + len(self._current_content)
-
-        # Adjust the start based on any data read past the current content
-        start += (self._offset - len(self._current_content))
-        return start
-
     def chunks(self):
         # type: () -> Iterator[bytes]
-        """Iterate over chunks in the download stream.
+        """
+        Iterate over chunks in the download stream. Note, the iterator returned will
+        iterate over the entire download content, regardless of any data that was
+        previously read.
```

Comment:
It could be worth adding a warning here that calling this on a partially read stream may result in some re-fetching of data, so people will be aware of the inefficiencies is using it that way.

Reply:
Yep, this is how it's always been since we added the ability to partially read the stream. The only main difference is we used to always buffer the first chunk, now it can be unloaded. I've added a warning in the case where we know we will need to re-fetch data.

Reply:
Ah - sorry poor choice of words on my part here - by "warning" here, I just meant adding it to the docstring, maybe with the bold: **warning** or **note** tags - I don't think we need a runtime logged warning for this.

Reply:
Ah, ok. Removed the runtime warning and added a note to the docstring instead. Thanks